data-prepper-plugins/common/src/main/java/org/opensearch/dataprepper/plugins/accumulator/Buffer.java

```diff
@@ -0,0 +1,28 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.dataprepper.plugins.accumulator;
+
+import software.amazon.awssdk.services.s3.S3Client;
+
+import java.io.IOException;
+
+/**
+ * A buffer can hold data before flushing it to S3.
+ */
+public interface Buffer {
+
+    /**
+     * Gets the current size of the buffer. This should be the number of bytes.
+     * @return buffer size.
+     */
+    long getSize();
+    int getEventCount();
+
+    long getDuration();
+
+    void flushToS3(S3Client s3Client, String bucket, String key) ;
```

Comment:
This is specific to S3 sink, not something common. I think you will still need `sendDataToHttpEndpoint` for http sink.

Reply:
Thanks for the review @oeyh , Added sendDataToHttpEndpoint() in Buffer

Reply:
Thanks for the review @oeyh, We have removed the methods and shall implement in the respective plugins.